packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1014,16 +1009,19 @@ export default abstract class SqlIntegration
           ${isRatio ? `d` : `m`}.variation,
           ${isRatio ? `d` : `m`}.dimension,
           ${this.ensureFloat("COUNT(*)")} as count,
-          ${this.avg("m.value")} as m_mean,
-          ${this.variance("m.value")} as m_var,
+          ${this.avg("coalesce(m.value,0)")} as m_mean,
+          ${this.variance("coalesce(m.value,0)")} as m_var,
           ${this.ensureFloat("sum(m.value)")} as m_sum
           ${
             isRatio
               ? `,
-            ${this.avg("d.value")} as d_mean,
-            ${this.variance("d.value")} as d_var,
+            ${this.avg("coalesce(d.value,0)")} as d_mean,
```

Comment:
If we make the above one line edit, and we keep this `LEFT JOIN` for ratio metrics, then we only need coalesce on the m.values, not on the d.values. Of course, if we change to `FULL OUTER JOIN` for ratio metrics, then we would need both coalesce statements.

Reply:
Yeah, I didn't want to rely on the specific joins we're using today since that is likely to change in the future.